### ISSUE TITLE
TypeError: 'NoneType' object is not callable

### ISSUE THREAD

**Issue by lukas-vlcek** [BitScoutOrg/efk-atomicapp#16]
Using atomic 1.8 the build fails with:

```
2016-02-05 08:34:36,880 - atomicapp.nulecule.container - INFO - Extracting nulecule data from image: viaq/nulecule-rsyslog-collector to /host/var/lib/atomicapp/viaq-efk-atomicapp-2ec6d72702d3/external/viaq-nulecule-rsyslog-collector
2016-02-05 08:34:36,880 - atomicapp.nulecule.container - DEBUG - Creating docker container: /usr/bin/docker create --entrypoint /bin/true viaq/nulecule-rsyslog-collector
2016-02-05 08:34:37,281 - atomicapp.nulecule.container - DEBUG - Copying data from Docker container: /usr/bin/docker cp 08befe2d141c736fbc2ce9621ae6e84908e5d589f2b523285e7200834b335a15:/application-entity /tmp/nulecule-4a836ffe-cbe3-11e5-8ca0-f0def1f9b055
2016-02-05 08:34:37,425 - atomicapp.nulecule.container - DEBUG - Copying nulecule data from /tmp/nulecule-4a836ffe-cbe3-11e5-8ca0-f0def1f9b055 to /host/var/lib/atomicapp/viaq-efk-atomicapp-2ec6d72702d3/external/viaq-nulecule-rsyslog-collector
2016-02-05 08:34:37,427 - atomicapp.nulecule.container - DEBUG - Removing tmp dir: /tmp/nulecule-4a836ffe-cbe3-11e5-8ca0-f0def1f9b055
2016-02-05 08:34:37,427 - atomicapp.utils - DEBUG - Recursively removing directory: /tmp/nulecule-4a836ffe-cbe3-11e5-8ca0-f0def1f9b055
2016-02-05 08:34:37,428 - atomicapp.nulecule.container - DEBUG - Removing Docker container: /usr/bin/docker rm -f 08befe2d141c736fbc2ce9621ae6e84908e5d589f2b523285e7200834b335a15
08befe2d141c736fbc2ce9621ae6e84908e5d589f2b523285e7200834b335a15
2016-02-05 08:34:38,059 - atomicapp.plugin - DEBUG - Loading providers from /opt/atomicapp/atomicapp/providers
2016-02-05 08:34:38,060 - atomicapp.plugin - DEBUG - Loading providers from /opt/atomicapp/atomicapp/providers
2016-02-05 08:34:38,065 - __main__ - ERROR - 'NoneType' object is not callable
Traceback (most recent call last):
  File "/opt/atomicapp/atomicapp/cli/main.py", line 100, in cli_run
    nm.run(**argdict)
  File "/opt/atomicapp/atomicapp/nulecule/main.py", line 236, in run
    self.nulecule.run(cli_provider, dryrun)
  File "/opt/atomicapp/atomicapp/nulecule/base.py", line 148, in run
    provider_key, provider = self.get_provider(provider_key, dryrun)
  File "/opt/atomicapp/atomicapp/nulecule/lib.py", line 99, in get_provider
    self.get_context(), self.basepath, dry)
TypeError: 'NoneType' object is not callable
```

**Comment by lukas-vlcek** [comment]
Surprisingly, the reason is presence of `README` file in `artifacts` directory. When this file is delete the build completes successfully.

**Comment by richm** [comment]
And the reason there was a README file there was because 2 weeks ago it failed if I tried to use an empty artifacts directory . . . it is very frustrating that nulecule is so unstable 

Sent from my Verizon Wireless 4G LTE smartphone

-------- Original message --------

**Comment by lukas-vlcek** [comment]
But this time the situation is little different. It really requires the `artifacts` directory to be there. However, I do not know how to fix it now. Empty directory can not be part of git repo.
- I tried to replace the `README` file with `.gitignore` but it leads to the same error.
- I tried to remove the `artifacts` directory and create it in `Dockerfile` (before `ADD` commands). It fails too.
- I tried to delete all files from `artifacts` directory in `Dockerfile` (before `ADD` commands). It fails too.

I am running out of ideas...
